pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -290,6 +293,11 @@ def transfer_distrib_spawn_state_on_fit_end(self, results):
             self.mp_queue.put(best_model_path)
             self.mp_queue.put(last_path)
             self.mp_queue.put(results)
+            self.mp_queue.put({
+                "callback_metrics": apply_to_collection(
+                    self.lightning_module.trainer.logger_connector.callback_metrics, torch.Tensor, lambda x: x.item()
```

Comment:
do you know why we need to call x.item()?

Reply:
there was an issue when pickling cuda tensors. We discussed with @tchaton to cast tensors to primitive float types

Reply:
and doing .cpu() instead of .item(), have you thought about it?

Reply:
@edgarriba I also suggest using a cpu-tensor instead of primitives if possible. This means that we can still use PyTorch's multiprocessing package (depending on shared memory instead of pickling) and we avoid the tensor creation on the main proc (IMO the type has to be restored otherwise)